test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetSourcesCommandTest.cs

```
@@ -411,6 +411,191 @@ public void SourcesCommandTest_AddWithUserNamePassword_UserDefinedConfigFile()
             }
         }
 
+        [Theory]
+        [InlineData(@"https://source.test", true)]
+        [InlineData(@"\\myserver\packages", false)]
+        public void SourcesCommandTest_AddWithProtocolVersion(string source, bool shouldWriteProtocolVersion)
```

This test failed on Mac (we don't run Mono tests on Linux):

> Assert.Equal() Failure
> Expected: (null)
> Actual:   3

The test has me wondering if the behaviour is good though. On one hand `protocolVersion` isn't used for local feeds, so I understand not using it for local feeds. On the other hand, if a customer passes `-ProtocolVersion 3` on the command line, I don't like that it gets silently ignored. To me, an error message saying "local feeds don't support protocolVersion" feels better, as it's more explicit about what's going on.

---

I've pushed a change that should fix tests failing on Mac. Switched to using `Util.VerifyResultSuccess` which is used in other tests instead of `VerifyResultFailure`.

As for behaviour with local feeds, I've implemented it as per [this comment](https://github.com/NuGet/NuGet.Client/pull/5002#pullrequestreview-1287837289).

---

I believe that the reason why test failed on Mac was due to `\\myserver\packages` not being treated as local source. I've split the test into two: one for remote, one for local. Test for local source uses different source path depending on the platform it runs on.

---

My bad! I updated only tests for `update` and forgot about `add`. I've just pushed updated tests for `add`.